main.py

```diff
@@ -0,0 +1,35 @@
+import random
+from monster import Monster
+from hero import Hero
+
+
+def choose_action(hero: Hero) -> None:
+    action = input(
+        "enter action: \n'a' - (attack) \n'u' - (upgrade) \n'r' - (revive) \n'd' - (defense) \n"
+    )
+    if action == "a":
+        hero.attack(monster1)
+    elif action == "u":
+        hero.level_up()
+    elif action == "r":
+        hero.heal()
+    elif action == "d":
+        hero.defend()
+
+    ezra.increase_coins(1)
+
+
+if __name__ == "__main__":
+    hero_dead = False
+    ezra = Hero("ezra")
+    monster1 = Monster("m1", random.choice([ezra.level - 1, ezra.level + 1]))
```

Comment:
if your level is 1, the monster level could be 0.           
the randomise of the level should be done in the constructor of monster